src/libmongoc/tests/unified/operation.c

```diff
@@ -1546,13 +1656,20 @@ operation_upload (test_t *test,
       goto done;
    }
 
+   opts = bson_copy (bson_parser_get_extra (bp));
+   if (op->session) {
+      if (!mongoc_client_session_append (op->session, opts, error)) {
```

Comment:
GridFS does not support sessions. See [SPEC-921](https://jira.mongodb.org/browse/SPEC-921) and [DRIVERS-559](https://jira.mongodb.org/browse/DRIVERS-559).

As a rule of thumb, I think you should double-check that `mongoc_client_session_append` is not being introduced for any methods that aren't documented as taking a "sessionId" parameter. In this case, [`mongoc_gridfs_bucket_open_upload_stream`](http://mongoc.org/libmongoc/current/mongoc_gridfs_bucket_open_upload_stream.html) makes no mention of sessions.

Reply:
Yeah... I should have taken a look at the documentation for each. Removed the session appension here, in `assert_collection_exists` and `assert_index_exists`.